packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeRenderer.tsx

```diff
@@ -46,20 +45,47 @@ export function TreeRenderer({ rootNodes, expandNode, localizedStrings, selectNo
     selectionMode: selectionMode ?? "single",
   });
 
+  const flatNodes = useMemo(() => getFlatNodes(rootNodes, 0), [rootNodes]);
+
   return (
     <LocalizationContextProvider localizedStrings={localizedStrings}>
       <Tree.Root style={{ height: "100%", width: "100%" }}>
-        <TreeLevelRenderer
-          {...treeProps}
-          nodes={rootNodes}
-          expandNode={expandNode}
-          onNodeClick={onNodeClick}
-          onNodeKeyDown={onNodeKeyDown}
-          isNodeSelected={isNodeSelected}
-        />
+        {flatNodes.map((flatNode) => (
+          <TreeNodeRenderer
+            {...treeProps}
+            aria-level={flatNode.level}
+            aria-posinset={flatNode.posInLevel}
+            aria-setsize={flatNode.levelSize}
+            expandNode={expandNode}
+            onNodeClick={onNodeClick}
+            onNodeKeyDown={onNodeKeyDown}
+            node={flatNode.node}
+            key={flatNode.node.id}
+            selected={isNodeSelected?.(flatNode.node.id)}
+          />
+        ))}
       </Tree.Root>
     </LocalizationContextProvider>
   );
 }
 
 function noopSelectNodes() {}
+
+interface FlatPresentationTreeNode {
+  node: PresentationTreeNode;
+  level: number;
+  levelSize: number;
+  posInLevel: number;
+}
+
+export function getFlatNodes(nodes: PresentationTreeNode[], level: number) {
+  const flatNodes: FlatPresentationTreeNode[] = [];
+  nodes.map((node, index) => {
```

Comment:
`.forEach` is more applicable in this use case.

Reply:
Agree, changed